e2e_test/batch/join/asof_join.slt

```diff
@@ -0,0 +1,40 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+statement ok
+create table t1 (v1 int, v2 int, v3 int primary key);
+
+statement ok
+create table t2 (v1 int, v2 int, v3 int primary key);
+
+statement ok
+insert into t1 values (1, 2, 3), (2, 3, 4);
+
+statement ok
+insert into t2 values (1, 3, 4), (1, 2, 5), (1, 2, 6);
+
+# asof inner join
+query IIIIII
+SELECT t1.v1 t1_v1, t1.v2 t1_v2, t1.v3 t1_v3, t2.v1 t2_v1, t2.v2 t2_v2, t2.v3 t2_v3 FROM t1 ASOF JOIN t2 ON t1.v1 = t2.v1 and t1.v2 < t2.v2 order by t1.v1;
+----
+1 2 3 1 3 4
```

Comment:
Add one more row to the t1 and run this sql again. 
```sql
 insert into t1 values (1, 2, 9);
```

The expected result should be (verified by duckdb)
```
┌───────┬───────┬───────┬───────┬───────┬───────┐
│ t1_v1 │ t1_v2 │ t1_v3 │ t2_v1 │ t2_v2 │ t2_v3 │
│ int32 │ int32 │ int32 │ int32 │ int32 │ int32 │
├───────┼───────┼───────┼───────┼───────┼───────┤
│     1 │     2 │     3 │     1 │     3 │     4 │
│     1 │     2 │     9 │     1 │     3 │     4 │
└───────┴───────┴───────┴───────┴───────┴───────┘
```

but current implementation will give us
```
 t1_v1 | t1_v2 | t1_v3 | t2_v1 | t2_v2 | t2_v3
-------+-------+-------+-------+-------+-------
     1 |     2 |     3 |     1 |     3 |     4
(1 row)
```